app/services/name_generator.rb

```diff
@@ -0,0 +1,37 @@
+class NameGenerator
+  def initialize(type = nil)
+    @type = type || Setting[:name_generator_type]
+    @mac_generator = Deacon::MacGenerator.new
+    @random_generator = Deacon::RandomGenerator.new
+  end
+
+  def next_mac_name(mac)
+    if @type =~ /^MAC/i && mac
```

Comment:
How about extracting this to a (class) method, `if random_based? && mac`, you could use `delegate :random_based?, to: self.class` to make this a little cleaner imho.